### DISCUSSION TITLE
For each unknown TS

### DISCUSSION THREAD

**Discussion by wommy** [solidjs/solid#867] [General]
super new to typescript, had vscode yelling at me,

```tsx 
<For each={todos().reverse()} fallback={<p>loading...</p>}>
  {todo => <li>{todo}/*<~ error here */</li>}
</For>
```

i fixed the typeof unknown  by adding `as string`

```diff
<For each={todos().reverse()} fallback={<p>loading...</p>}>
-  {todo => <li>{todo}</li>}
+  {todo => <li>{todo as string}</li>}
</For>
```

**Comment by untlsn** [comment]
What type is the todo? Maybe it's not a string?
If you hover over todo() you should see Accessor with todo type inside <>

**Reply by wommy** [reply]
i really appreciate the help, my TS knowledge is pretty limited,

im hitting the sanity API with their official [JS client](https://www.sanity.io/docs/js-client#performing-queries)
im guessing im either not awaiting something correctly, or maybe its a weird interface issue with sanity

ive tried to slim down the code for clarity, might have messed something up on the way,

```tsx
import { Component, createResource, For } from 'solid-js'
import client from '@sanity/client'
import auth from './auth'

export const Todo: Component = () => {
  const [todos] = createResource(async () =>
    client(auth).fetch(`*[_type=='todo'] | order(_createdAt desc).todo`),
  )
  return <For each={todos()}>{ea => <li>{ea}</li>}</For>
}

```

my list generates everything, just i have this one Error on the returned `{ea}` in the `<li>`

```
(parameter) ea: unknown
Type 'unknown' is not assignable to type 'Element'.
  Type 'unknown' is not assignable to type 'string & {}'.
    Type 'unknown' is not assignable to type 'string'.ts(2322)
[jsx.d.ts(80, 5): ]()The expected type comes from property 'children' which is declared here on type 'LiHTMLAttributes<HTMLLIElement>'
```

besides the error above, ill list the hovers for everything else in the following block

#### `return <For each={todos()}>{ea => <li>{ea}</li>}</For>`

- [ ] `{ea =>`: `(parameter) ea: unknown`
- [ ] `{todos()}`: `const todos: Resource`, `() => any`
- [ ] `each=`: `(JSX attribute) each: false | readonly unknown[]`

i appreciate the reply!

the more im thinking about this, its probably something with the sanity return,
im gonna try a vanilla fetch to their api next

**Reply by wommy** [reply]
yeah 99% sure this is from sanity

**Reply by wommy** [reply]
yup 200%,

how do i resolve this
